### ISSUE TITLE
Unsupported operand types for - ("KeysView[str]" and "Tuple[str, ...]")

### ISSUE THREAD

**Issue by vr2262** [python/typeshed#2919]
I believe this is similar to #2454 

For this `example.py`:
```python
from typing import AbstractSet, Any, Dict, Tuple

def func(a: Dict[str, Any], b: Tuple[str, ...]) -> AbstractSet:
    return a.keys() - b
```

The function works:
```
$ python -c "from example import func; print(func(dict(a=1, b=2), ('a',)))"
{'b'}
```

But `mypy` doesn't like it:
```
$ python -m mypy example.py
example.py:4: error: Unsupported operand types for - ("KeysView[str]" and "Tuple[str, ...]")
```

**Comment by srittau** [comment]
Patches welcome!

**Comment by vr2262** [comment]
I looked into it a little further and found that all of these operations are supported but don't pass the type check:
```python
from typing import AbstractSet, Any, Dict, KeysView, ItemsView, Tuple, Union

def dict_gauntlet(kv: Dict) -> None:
    other = ('a', 2)

    # This appears to be a known issue with AbstractSet
    a = kv.keys().isdisjoint(other)
    print(a)
    b = kv.items().isdisjoint(other)
    print(b)

    c: AbstractSet[Any] = kv.keys() | other
    print(c)
    d: AbstractSet[Any] = kv.items() | other
    print(d)

    e: AbstractSet[Any] = kv.keys() & other
    print(e)
    f: AbstractSet[Any] = kv.items() & other
    print(f)

    g: AbstractSet[Any] = kv.keys() - other
    print(g)
    h: AbstractSet[Any] = kv.items() - other
    print(h)

    i: AbstractSet[Any] = kv.keys() ^ other
    print(i)
    j: AbstractSet[Any] = kv.items() ^ other
    print(j)


dict_gauntlet({1: 1, 'a': 'a'})
```

I tried making changes to `stdlib/3/typing.pyi` but I'm not sure how to make `mypy` take them into account, or write tests for them.

**Comment by Hanaasagi** [comment]
@vr2262 
I think I have fixed some in #2906. But I didn't notice that some methods(e.g. `__sub__`) are missing here. It's my fault. If you are willing to fix it, please pull request.

If you want to make `mypy` take them into account, you can try `pip show mypy` and find the install location, then replcace `typeshed` in `mypy` directory.

**Comment by vr2262** [comment]
@srittau @Hanaasagi I opened #2997 to fix this. Thanks for your help!
